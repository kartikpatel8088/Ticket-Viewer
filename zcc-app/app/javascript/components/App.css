.header {
  text-align: center;
}

.tag-badge{
  font-color: white;
}

.logo{
  width: 200px;
  height: 150px;
}

.via{
  font-size: 15px;
  padding-top: 10px;
  color: grey;
}