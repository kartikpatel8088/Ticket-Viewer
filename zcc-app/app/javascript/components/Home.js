import React, { useState } from 'react';

import Toast from 'react-bootstrap/Toast';
import Container from 'react-bootstrap/Container';
import Button from 'react-bootstrap/Button';
import Image from 'react-bootstrap/Image';
import Table from 'react-bootstrap/Table';
import Row from 'react-bootstrap/Row';
import Logo from '../../assets/images/zendesk_logo.png';


import './App.css';

function Home(){

	return	(
			
			<Container className="p-4">
			    <Container className="p-5 mb-4 bg-light rounded-3">
			    	<Container>
			    	<Row className="justify-content-md-center">
			      	<Image src={Logo} rounded className="logo" />
			      	</Row>
			      	</Container>
			      	<br />	
			      <h5 className="header">Welcome to the Zendesk Coding Challenge - Ticket Viewer</h5>
			      <p className=""><u>Technologies</u> <br />Frontend-<b>React</b><br /> Backend- <b>Ruby on Rails</b><br /> VCS- <b>Github</b><br /> Responsive- <b>Bootstrap</b> </p>


			    </Container>
			  	
			</Container>	
			
		)


}


export default Home;
