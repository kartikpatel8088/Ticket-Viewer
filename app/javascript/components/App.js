import React from 'react'
import Home from './Home'
import TicketList from './TicketList';


function App(){

	return	(
			
			<div>
				<Home />
				<TicketList />
			</div>
			
		)

}

export default App

