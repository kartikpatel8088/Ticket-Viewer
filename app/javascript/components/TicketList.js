import React,{useEffect, useState} from 'react'
import Home from './Home'
import TicketDetails from './TicketDetails'
import './App.css'

import Table from 'react-bootstrap/Table';
import Container from 'react-bootstrap/Container';
import ProgressBar from 'react-bootstrap/ProgressBar';
import Badge from 'react-bootstrap/Badge';
import Spinner from 'react-bootstrap/Spinner';
import Button from 'react-bootstrap/Button';

import BootstrapTable from "react-bootstrap-table-next";
import paginationFactory from "react-bootstrap-table2-paginator";
import moment from 'moment';

// TicketList Class based Component - Stateful

class TicketList extends React.Component {
	constructor(props) {
	    super(props);
	    this.state = {
	      data: [],
	      ticketid: '',
	      show: false,
	      loading: true
	    };
	}

	componentDidMount() {
    	// Runs after the first render() lifecycle
    	const url = "http://localhost:3000/api/v1/base";
	    	fetch(url, {
			  "method": "GET",
			})
			.then(response => response.json())
			.then(response => {
			  this.setState({data: response.tickets}); 
			  this.setState({loading: false});
			})
			.catch(err => { console.log(err); 
			})
	};

	render(){
		function TagFormatter(cell, row){


			
			var tag = [];
			for (var j = 0; j < cell.length; j++) {
				
				
				tag.push(<Badge pill bg="secondary" text="light"> { cell[j] } </Badge>);
				/// 
			     
			}
			// console.log(tag)
			return (
				tag 
				
			);
		};

		function DateFormatter(cell, row){

			return	(
				moment(cell).format("MMM Do YYYY, h:mm:ss a")
			)
		};

		const columns = [
			{
				dataField: "id",
				text:"ID",
				sort: true	
			},
			{
				dataField: "subject",
				text:"Subject",
				sort: true
			},
			{
				dataField: "status",
				text:"Status",
				sort: true
			},
			{
				dataField: "tags",
				text:"Tags",
				sort: true,
				formatter: TagFormatter,
			},
			{
				dataField: "created_at",
				text:"Created",
				sort: true,
				formatter: DateFormatter,
			},
			{
				dataField: "updated_at",
				text:"Updated",
				sort: true,
				formatter: DateFormatter,
			},
		];

		const sizePerPageList = [
			{
			  text: '5', value: 5
			}, {
			  text: '15', value: 15
			}, {
			  text: '25', value: 25
			},
			{
			  text: 'All', value: this.state.data.length
			}  
		];

		const ticketEvent = {

			onClick: (e, row, rowIndex) => {
			     this.setState({ticketid: row.id});
			     this.setState({show: true});
			 

			},
		};

		const fromTicketDetail = (childdata) => {
		    this.setState({show: childdata});
		 }
		

		return (
			!this.state.loading ? 
				<>
					
					<Container>	
					 <BootstrapTable hover
				        bootstrap4
				        keyField="id"
				        data={this.state.data}
				        columns={columns}
				        rowEvents={ ticketEvent }
				        pagination={paginationFactory({ sizePerPage: 25 , sizePerPageList: sizePerPageList})}
				      />
					
					</Container>

					{this.state.show && <TicketDetails id={this.state.ticketid} show={this.state.show} onHide={() => this.setState({show: false})}/>}	
				</>
				:

				<Container className="header">
					<Button variant="primary" disabled>
				    <Spinner
				      as="span"
				      animation="grow"
				      size="sm"
				      role="status"
				      aria-hidden="true"
				    />
				    {" "}
				    Loading...
				  </Button>
					
				</Container>


		)
	}
}


/// TicketDetails Functional based Component - Stateless 

// function TicketList() {
// 	const [loading, setLoading] = React.useState(true);
// 	const [show, setShow] = React.useState(false);
// 	const [data, setData] = React.useState([]);
// 	const [ticketid, setId] = React.useState();

// 	function TagFormatter(cell, row) {

// 		var tag = []
// 		for (var i = 0; i < cell.length; i++) {
// 		    tag.push(<Badge pill bg="secondary" text="light">{ cell[i] } </Badge> );
// 		}
// 		return (
// 			tag
// 		);
// 	}

// 	function DateFormatter(cell, row) {

// 		return	(
// 			moment(cell).format("MMM Do YYYY, h:mm:ss a")
// 		)
// 	}

// 	const columns = [
// 		{
// 			dataField: "id",
// 			text:"ID",
// 			sort: true	
// 		},
// 		{
// 			dataField: "subject",
// 			text:"Subject",
// 			sort: true
// 		},
// 		{
// 			dataField: "status",
// 			text:"Status",
// 			sort: true
// 		},
// 		{
// 			dataField: "tags",
// 			text:"Tags",
// 			sort: true,
// 			formatter: TagFormatter,
// 		},
// 		{
// 			dataField: "created_at",
// 			text:"Created",
// 			sort: true,
// 			formatter: DateFormatter,
// 		},
// 		{
// 			dataField: "updated_at",
// 			text:"Updated",
// 			sort: true,
// 			formatter: DateFormatter,
// 		},

// 	]

// 	const sizePerPageList = [
// 		{
// 		  text: '5', value: 5
// 		}, {
// 		  text: '15', value: 15
// 		}, {
// 		  text: '25', value: 25
// 		},
// 		{
// 		  text: 'All', value: data.length
// 		}  
// 	]

// 	const ticketEvent = {

// 		onClick: (e, row, rowIndex) => {
// 		     // console.log(`clicked on row with index: ${row.id}`);
// 		     setId(row.id);
// 		     setShow(true);

// 		},
// 	}

// 	const fromTicketDetail = (childdata) => {
// 	    setShow(childdata);
// 	 }

// 	useEffect(()=>{

// 		const url = "http://localhost:3000/api/v1/base";
// 	    	fetch(url, {
// 			  "method": "GET",
// 			})
// 			.then(response => response.json())
// 			.then(response => {
// 			  setData(response.tickets); 
// 			  setLoading(false);
// 			})
// 			.catch(err => { console.log(err); 
// 			})

// 	}, []);

// 	return	!loading ? 
// 			<>
				
// 				<Container>	
// 				 <BootstrapTable hover
// 			        bootstrap4
// 			        keyField="id"
// 			        data={data}
// 			        columns={columns}
// 			        rowEvents={ ticketEvent }
// 			        pagination={paginationFactory({ sizePerPage: 25 , sizePerPageList: sizePerPageList})}
// 			      />
				
// 				</Container>

// 				{show && <TicketDetails id={ticketid} show={show} onHide={() => setShow(false)}/>}	
// 			</>
// 			:

// 			<Container className="header">
// 				<Button variant="primary" disabled>
// 			    <Spinner
// 			      as="span"
// 			      animation="grow"
// 			      size="sm"
// 			      role="status"
// 			      aria-hidden="true"
// 			    />
// 			    {" "}
// 			    Loading...
// 			  </Button>
				
// 			</Container>

// }


export default TicketList