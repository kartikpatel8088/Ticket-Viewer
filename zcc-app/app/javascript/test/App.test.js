
import React from "react";
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


import App from "./../components/App";
import Home from './../components/Home';
import TicketList from './../components/TicketList';
import 'isomorphic-fetch';

configure({ adapter: new Adapter() });



describe('<App />', () => {
  it('should renders initial <App /> components', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.getElements()).toMatchSnapshot();

  });

  it('<App /> component renders <Home /> component', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toContainReact(<Home />);
  });

  it('<App /> component renders <TicketList /> component', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toContainReact(<TicketList />);
  });

});


