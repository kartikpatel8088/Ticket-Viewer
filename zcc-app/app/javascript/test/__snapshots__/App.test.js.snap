// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should renders initial <App /> components 1`] = `
Array [
  <div>
    <Home />
    <TicketList />
  </div>,
]
`;
